langchain4j-core/src/main/java/dev/langchain4j/model/discovery/ModelDescription.java

```diff
@@ -0,0 +1,319 @@
+package dev.langchain4j.model.discovery;
+
+import dev.langchain4j.model.ModelProvider;
+import dev.langchain4j.model.chat.Capability;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents metadata about an available model from a provider.
+ * This class provides a unified view of model information across different providers.
+ *
+ * <p>Only {@code id}, {@code name}, and {@code provider} are required fields.
+ * All other fields are optional and may be null depending on what information
+ * the provider makes available.
+ */
+public class ModelDescription {
+
+    private final String id;
+    private final String name;
+    private final String description;
+    private final ModelProvider provider;
+    private final ModelType type;
+    private final Set<Capability> capabilities;
+    private final ModelPricing pricing;
+    private final Integer contextWindow;
```

Comment:
```suggestion
    private final Integer maxTotalTokens;
```
Please also move it after `maxInputTokens` and `maxOutputTokens`

Reply:
I removed this, because it was de-facto only used for exactly providing the maxInputTokens - only for Gemini.